### ISSUE TITLE
searching when no database is selected in a project crashes bw2-browser

### ISSUE THREAD

**Issue by tngTUDOR** [brightway-lca/brightway2-ui#10]
**[Original report](https://bitbucket.org/tomas_navarrete/brightway2-ui/issues/10) by Anonymous.**

----------------------------------------

I you start bw2-browser, select a project and immediately "search" for an activity, the browser crashes with:

```
Projects

[0]: default
[1]: universe

Page 0 of 1. Use n (next page) and p (previous page) to navigate.

>> 1

Databases

[0]: biosphere3 (4321 activities/flows)
[1]: universe-cutoff (42 activities/flows)

Page 0 of 1. Use n (next page) and p (previous page) to navigate.

ecoinvent36 >> s meaning of life
Traceback (most recent call last):
  File "/opt/conda/bin/bw2-browser", line 10, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.9/site-packages/bw2ui/bin/bw2_browser.py", line 1337, in main
    activitybrowser.cmdloop()
  File "/opt/conda/lib/python3.9/cmd.py", line 138, in cmdloop
    stop = self.onecmd(line)
  File "/opt/conda/lib/python3.9/cmd.py", line 217, in onecmd
    return func(arg)
  File "/opt/conda/lib/python3.9/site-packages/bw2ui/bin/bw2_browser.py", line 1024, in do_s
    results = Database(self.database).search(needle, limit=self.search_limit)
  File "/opt/conda/lib/python3.9/site-packages/bw2data/backends/peewee/database.py", line 530, in search
    with Searcher(self.filename) as s:
  File "/opt/conda/lib/python3.9/site-packages/bw2data/backends/base.py", line 122, in filename
    return safe_filename(self.name)
  File "/opt/conda/lib/python3.9/site-packages/bw2data/filesystem.py", line 32, in safe_filename
    return safe + u"." + hashlib.md5(string).hexdigest()
TypeError: object supporting the buffer API required

```

**Comment by tngTUDOR** [comment]
**Original comment by Tomas Navarrete Gutierrez (Bitbucket: [tomas_navarrete](https://bitbucket.org/tomas_navarrete), ).**

----------------------------------------

🐛 Fix #10 - crash when searching without active db

**Comment by tngTUDOR** [comment]
**Original comment by Tomas Navarrete Gutierrez (Bitbucket: [tomas_navarrete](https://bitbucket.org/tomas_navarrete), ).**

----------------------------------------

🐛 Fix #10 - crash when searching without active db
